### PR TITLE
Renaming base url setting

### PR DESCRIPTION
**Pull request by sandilsranasinghe**
Changing BASE_URL to WAGTAILADMIN_BASE_URL according to #3248

- `settings.BASE_URL` changed to `settings.WAGTAILADMIN_BASE_URL`
- falls back to `request.site.root_url` or the host name of the request if in a request context
- added documentation for WAGTAILADMIN_BASE_URL
- updated usage of `settings.BASE_URL` in html templates to use a tag (`base_url_setting`) instead
- pulled out code for getting the base_url from the tag `base_url_setting` to a separate function `get_admin_base_url`

### CODE REVIEW

**Comment by lb-** [comment]
@sandilsranasinghe thanks for submitting this PR! Really appreciate it. 
Please mark this as a non-draft when you feel it is ready to be reviewed, otherwise it will not get a review from the core team.

**Comment by sandilsranasinghe** [comment]
sorry about the delay with the draft, I wanted to make a couple more changes

**Comment by lb-** [comment]
No problems, just wanted to make sure your setting to draft was intentional.
I have triggered the CI workflow for this.

File: wagtail/test/settings.py
Comment (lb-) [inline wagtail/test/settings.py:8]: we should not need this commented out code

File: wagtail/admin/utils.py
Comment (lb-) [inline wagtail/admin/utils.py:14]: We may want to add a deprecated warning here, but only if `WAGTAILADMIN_BASE_URL` is `None` and `BASE_URL` is not `None`.

See `wagtail/utils/deprecation.py` for the utils we use for this.

This would probably be a two release warning, so I recommend you use `RemovedInWagtail50Warning`, you can search through the code for other examples.

File: docs/reference/settings.rst
Comment (lb-) [inline docs/reference/settings.rst:56]:
```suggestion
If this setting is not present, Wagtail will try to fall back to ``request.site.root_url`` or to the requests host name.
```
Just a full stop at the end.

File: docs/reference/settings.rst
Comment (lb-) [inline docs/reference/settings.rst:54]:
```suggestion
This is the base URL used by the Wagtail admin site. It is typically used for generating URLs to include in notification emails.
```
Remove the previous name here, we will add that below (see other comment), plus a small fix for the usage note.

File: docs/reference/settings.rst
Comment (lb-) [inline docs/reference/settings.rst:57]: We can add a RST syntax that helps for changes
```
.. versionchanged:: 2.15

This setting was previously named ``BASE_URL`` and was undocumented, using ``BASE_URL`` will be removed in a future release.
```

**Review by lb- — CHANGES_REQUESTED** [review]
Looking great so far, a few recommended fixes and changes, I have not tested this locally though but feel free to ping me when these are updated.

Great stuff here! very thorough and thanks for pulling the logic out to a shared util.

File: docs/reference/settings.rst
Comment (lb-) [inline docs/reference/settings.rst:58]: sorry the 2.15 was just an example, the next release is 3.0
```suggestion
.. versionchanged:: 3.0
```

Comment (sandilsranasinghe) [reply docs/reference/settings.rst]: oops sorry didn't realize

File: docs/reference/settings.rst
Comment (lb-) [inline docs/reference/settings.rst:56]:
```suggestion
If this setting is not present, Wagtail will try to fall back to ``request.site.root_url`` or to the request's host name.
```

File: wagtail/admin/templates/wagtailadmin/notifications/base.txt
Comment (lb-) [inline wagtail/admin/templates/wagtailadmin/notifications/base.txt:6]:
```suggestion
{% base_url_setting as base_url %}
{% block content %}
```

File: wagtail/admin/templates/wagtailadmin/notifications/base.txt
Comment (lb-) [inline wagtail/admin/templates/wagtailadmin/notifications/base.txt:9]:
```suggestion
{% trans "Edit your notification preferences here: " %}{{ base_url }}{% url 'wagtailadmin_account' %}#notifications
```

**Review by lb- — CHANGES_REQUESTED** [review]
@sandilsranasinghe thanks for doing this, I have updated a bunch of other locations that were not using the template tag but I cannot seem to get this to work for me.

Please advise the steps you took to test this manually, maybe I am missing something but no matter what I do not get the base URL when testing (even when updating the setting in bakerydemo).

I only ever get the path after and the base url always resolves to nothing, and it appears the tag does not get called.

```Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
Subject: The page "Desserts with Benefits" has been submitted to workflow
 "Moderators
 approval"
From: webmaster@localhost
To: admin@example.com
Date: Tue, 05 Apr 2022 22:08:37 -0000
Message-ID: <164919651795.7539.7865116353604516530@ubuntu-impish>
Auto-Submitted: auto-generated

Hello Admin,


The page "Desserts with Benefits" has been submitted for moderation to workflow "Moderators approval" by Admin User.


You can edit the page here: /admin/pages/77/edit/



Edit your notification preferences here: /admin/account/#notifications
```

**Comment by sandilsranasinghe** [comment]
@lb-  my bad, it seems I wasn't thorough enough when testing it locally, it should be fixed now. 

I didn't realize that I had tested with `WAGTAILADMIN_NOTIFICATION_USE_HTML = True`, so I saw only the html templates while testing which I had updated with the tag, but the default is with text templates which I hadn't updated with the tag. I updated those with the tag now so it should be working fine.

In your case it wasn't working when you updated the other locations because you had a tiny syntax problem in how you called the tag. It should work with the changes I pushed now.

```Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
Subject: The page "Tracking Wild Yeast" has been submitted to workflow
 "Moderators
 approval"
From: webmaster@localhost
To: admin@example.com
Date: Wed, 06 Apr 2022 06:21:53 -0000
Message-ID: <164922611399.14871.9532375613167493793@sandil-Dell-G15-5511>
Auto-Submitted: auto-generated

Hello Admin,


The page "Tracking Wild Yeast" has been submitted for moderation to workflow "Moderators approval" by Admin User.


You can edit the page here: https://boogie.com/admin/pages/62/edit/



Edit your notification preferences here: https://boogie.com/admin/account/#notifications

```

thanks for taking your time to review the changes

**Comment by lb-** [comment]
@sandilsranasinghe  thanks so much, I will give this another go validating locally.

Sorry if I messed up your branch, I should have done a back up (or done the manual validation earlier).

**Review by lb- — APPROVED** [review]
@sandilsranasinghe working great, thanks for fixing this all up.
I have pushed to your branch (again) with the changelog and updated commit message.

I have added you to the contributors as `Sandil Ranasinghe` - please let me know if you wanted a different name listed.

I will let the CI run to double check but this is looking like I can merge in shortly.

Thanks so much and congrats on working through this feature to get your first contribution in Wagtail.

**Pull request merged**
